### PR TITLE
Add runner check in repo action page

### PR DESCRIPTION
**Pull request by yp05327**
image

Maybe we can also add online runner check? e.g. : Target runner is offline.

### CODE REVIEW

File: routers/web/repo/actions/actions.go [hunk 1]
```diff
@@ -86,6 +91,41 @@ func List(ctx *context.Context) {
...
+			runners, err := actions_model.FindRunners(ctx, opts)
+			if err != nil {
+				ctx.ServerError("FindRunners", err)
+				return
+			}
+			allRunnerLabels := make(container.Set[string])
+			for _, r := range runners {
+				allRunnerLabels.AddMultiple(r.AgentLabels...)
+				allRunnerLabels.AddMultiple(r.CustomLabels...)
+			}
```
Comment (wolfogre) [inline routers/web/repo/actions/actions.go]: There's no need to do this every loop.

File: routers/web/repo/actions/actions.go [hunk 2]
```diff
@@ -86,6 +91,41 @@ func List(ctx *context.Context) {
...
+			matchRunner := true
+			for _, v := range jobs {
+				id, job := v.Job()
+				runsOnList := job.RunsOn()
+				for _, ro := range runsOnList {
+					if !allRunnerLabels.Contains(ro) {
+						matchRunner = false
+						workflow.ErrMsg = id
+						break
```
Comment (wolfogre) [inline routers/web/repo/actions/actions.go:116]: Maybe break the outer `for`?

File: routers/web/repo/actions/actions.go [hunk 3]
```diff
@@ -24,9 +28,10 @@ const (
 )
 
 type Workflow struct {
-	Entry     git.TreeEntry
-	IsInvalid bool
-	ErrMsg    string
+	Entry              git.TreeEntry
+	IsInvalid          bool
+	HaveMatchingRunner bool
+	ErrMsg             string
```
Comment (wolfogre) [inline routers/web/repo/actions/actions.go]: How about combining them into `ErrMsg` and displaying the warning icon when it is not empty? Or we may need add more and more flags.

PS, I don't really think we need image when everything is fine, it's confusing, it looks like every job in the workflow resulted in success.

**Comment by Zettat123** [comment]
Just curious, why not show the runner's status in jobs list? IMO the `runs-on` is a part of a job, not the whole workflow file.

**Comment by wolfogre** [comment]
> Just curious, why not show the runner's status in jobs list? IMO the `runs-on` is a part of a job, not the whole workflow file.

I think the point is to help correct the workflow YAML file with some problems like invalid syntax or unsupported `runs-on`.

**Review by wolfogre — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**